resources/references/extension-reference/twig-function-reference.md

```diff
@@ -0,0 +1,35 @@
+# Shopware's twig functions
+
+In Shopware, we extend Twig's functionality by custom ones. See our own actions below.
+
+## Functions
+
+| Function | Description | --- |
+| --- | --- | --- | 
+| `sw_extends` | Inherits from another file with support for multi inheritance. The API is the same like in twigs default `extends` | --- | 
+| `sw_include` | Includes template partials with support for multi inheritance. The API is the same like in twigs default `include`
+| --- | 
+| `sw_icon` | Displays an icon from a given icon set| See [Add thumbnail](./../../../guides/plugins/plugins/storefront/use-media-thumbnails.md) guide for details.  | 
+| `sw_thumbnails` | Renders a <img> tag with correctly configured “srcset” and “sizes” attributes based on the provided parameters | See [Add custom icon](./../../../guides/plugins/plugins/storefront/add-icons.md) guide for more information. | 
+| `sw_csrf` | Generates a valid CSRF token and inject it as a hidden input field to the form | More details in [CSRF protection](./../../../guides/plugins/plugins/storefront/use-csrf-protection.md) guide.  | 
+
+## Filter
+
+| Filter | Description | Remarks | 
+| --- | --- |  --- |
+| `replace_recursive` | Enables recursive replacement in addition to twig's default `replace` filter |  --- | 
```

Comment:
Used here: `guides/plugins/plugins/storefront/add-custom-javascript.md`

Maybe link it as an example?